test/functional/ui/inccommand_spec.lua

```diff
@@ -1279,6 +1281,52 @@ describe(":substitute, inccommand=split", function()
     ]])
   end)
 
+  it("is fast when previewing", function()
+    -- do not disable inccomand if it is slow
+    command("set rdt=999999999")
+    -- Load a big file.
+    feed_command("silent edit! test/functional/fixtures/bigfile_manylines.txt")
+    command("set modifiable")
+
+    --[[
+    -- this is a benchmark, so here's a bit of ceremony
+    command("set inccommand=split")
+    for i = 1,9,1 do
+      feed(":%s/de"..tostring(i-1).."/de"..tostring(i).."/<Esc>")
+    end
+
+    command("set inccommand=")
+    for i = 1,9,1 do
+      command("%s/de"..tostring(i-1).."/de"..tostring(i).."/g")
+    end
+    ]]
+
+    -- now that the caches are warm, we measure full-file substitution vs.
+    -- a preview (which runs 4 times each time)
+    command("set inccommand=split")
+    local before_icm = reltime()
+    for i = 1,9,1 do
+      feed(":%s/de0/de"..tostring(i).."/<Esc>")
+      wait()
+    end
+    local time_icm = reltimefloat(reltime(before_icm))
+
+    --[[
+    command("edit!")
+    command("set inccommand=")
+    local before_subst = reltime()
+    for i = 1,9,1 do
+      command("%s/de"..tostring(i-1).."/de"..tostring(i).."/g")
+    end
+    local time_subst = reltimefloat(reltime(before_subst))
+
+    -- the substitution churns through 10k lines, while the preview only looks
+    -- at 12 (default of previewhight), so the following should hold on any
+    -- system, even with a lot of variation 
+    eq(time_icm, time_subst)
```

Comment:
Isn't this flaky even in a local environment?

Reply:
It's just for debugging stuff, I wanted to check for `time_icm * 10 < time_subst`, but `feed` isn't doing what I need (see below).